### DISCUSSION TITLE
modeling of resource extraction technologies

### DISCUSSION THREAD

**Discussion by SongminYu** [iiasa/message_ix#832] [Q&A]
While understanding the [fossil resource](https://github.com/iiasa/message_ix/blob/main/tutorial/westeros/westeros_fossil_resource.ipynb) tutorial, I found the concept of "resource" is a bit special. On one hand, it is a `commodity` and can serve as input for `technology`. On the other hand, it also has its own `volume` and `cost`, which is quite special:

- A normal `commodity` in MESSAGE does not have `volume` directly, but can only be calculated by multiplying the `ACT` of technologies and `output`.
- A normal `commodity` in MESSAGE does not have `cost`, which is the parameter of `technologies`.

Following this, I have two questions:

- First, I want to confirm if my understanding is right: `resource` is a special `commodity` on the edge of the system boundary and it is scarce in the system covered by the model, so it has to have `volume`, as well as `cost`. Are there any other considerations about why `resource` is designed in such a way?
- Second, is it possible to model resource extraction technologies (or mining sectors in general) in MESSAGE? If so, are there any suggestions?

**Comment by khaeru** [comment]
To your second question only, please note there is some methodological (but not code-focused) discussion in the docs of `message-ix-models` (for the IIASA family of global and country-scope models) on resources; see https://docs.messageix.org/projects/models/en/latest/global/energy/resource/fossilfuel.html and the following pages.

**Reply by SongminYu** [reply]
Hi @khaeru, thanks for your swift reply! So in MESSAGE, the convention is that the extraction technologies are embedded in the assumptions of `volume` and `cost` of the `resource`. They are not modeled explicitly. 

So, in a national model, taking `coal` as an example, can I have another technology `coal_import` --- only with `output` and `var_cost` meaning external import (as the last part of this [tutorial](https://github.com/iiasa/message_ix/blob/main/tutorial/westeros/westeros_multinode_energy_trade.ipynb)) --- to compete with `coal` extraction?

**Reply by khaeru** [reply]
- MESSAGE (the generic framework) lets one do many things. Your first question is about the generic framework and its many possible uses.
- "MESSAGEix-GLOBIOM" is our term for a particular set of models that use the generic MESSAGE features in a few particular ways. (I know the naming is confusing, sorry—I would have chosen something different.) I wouldn't call this "convention", just our way of using the framework. Some of the ways we currently use the framework are for legacy reasons and I would specifically counsel new users to avoid emulating.
- The link I gave above describes MESSAGEix-GLOBIOM, but per my comment it only describes the methods in a textual way, not specifically what is done in code.
- The other docs of `message-ix-models` describe **some** (not all) of the code/data that implement those methods.
- In particular, looking at [Technologies (`technology.yaml`)](https://docs.messageix.org/projects/models/en/latest/pkg-data/codelists.html#technologies-technology-yaml), there are many technology codes with `_extr_` in the ID.
- For example, there are `bio_extr_1` through `bio_extr_6`. I would assume (there are no specific docs on this point, so I cannot check) that these are used to represent different extraction technologies with different techno-economic characteristics that all (per the YAML file) `output: [biomass, primary]`.
- So I would say, *in the specific case of MESSAGEix-GLOBIOM*, that it's not true that "they are not modelled explicitly". They are.
- However I would guess that a modeler *could choose* to use the underlying/generic MESSAGE features to assign extraction costs (as opposed to other resource-related costs, like the costs of mineral rights or permits, etc.) to `resource_cost`. Nothing seems to prevent this.
- Maybe this is what happens in the fossil_resource tutorial; I didn't write it and am not so familiar, so can't say.

Per your new question, to take the example above, the `bio_extr_[123456]` technologies output to commodity=biomass, level=primary. Indeed any other technology, like an import/export technology, can be parameterized to `output` to the same (commodity, level), and would thus compete with `bio_extr_*`. However, I don't know how the `input` parametrization of `bio_extr_*` relate to the `resource_{cost,volume}` parametrization of coal or other commodities. I have asked colleagues who have worked directly with these details to look at your question here and chime in.
